build/migration/README.md

```diff
@@ -0,0 +1,98 @@
+
+``` yaml
+strategy:
+  matrix:
+    datastore: [ 'db2', 'postgres' ]
+      target: ['previous', 'last']
+```
+
+### Parameter: **datastore**
+
+The datastore that is being tested. 
+
+Each datastore layer that is tested as part of the framework uses the default build files and the files that match the `matrix.datastore` name added to the `migration.yml`.
+
+### Parameter: **target** 
+
+The target indicates the starting point of a release number or `previous` or last `release`. To keep this a low number of jobs, we should keep to the minimum numbers.
+
+Each release is specified by MAJOR.MINOR.PATCH. The framework tests using the previous MINOR release, and the previous release to the previous MINOR release. 
+
+Consider the following examples to see how the selection occurs:
+
+- **last** - The build looks back to the last tag.
+- **previous** - The build looks back to the tag PREVIOUS to the `last` tag's MINOR release. The previous tag is determined based on a lexigraphical sort, so any patch-build does not reset the order.
+- **MAJOR.MINOR.PATCH** - The build looks at the specific tag. e.g. 4.8.1
+
+Deisgn note, original implementation had a special parameter `Parameter: **type**`. The type indicates how the schema is reached... `step` by step or `direct` to the current `main` schema cli.
+
+## Files
+
+The following files are the files that orchestrate datastore specific tasks as well as common tasks.
+
+|Filename|Purpose|
+|----------|----------------|
+|`fhir/build/common/gather-environment-details.sh`|Gathers the environment details|
+|`fhir/build/migration/bin/0_determine.sh`|Determines if the build should run|
+|`fhir/build/migration/bin/1_check-cache.sh`|Checks if the file is cached and copied down|
+|`fhir/build/migration/bin/1_previous-setup.sh`|Runs the previous setup|
+|`fhir/build/migration/bin/2_compose.sh`|Setup docker compose and database used in previous release|
+|`fhir/build/migration/bin/2_previous-integration-test.sh`|Runs the previous integration tests|
+|`fhir/build/migration/bin/3_previous-teardown.sh`|Shuts down the previous server|
+|`fhir/build/migration/bin/3_previous-cache-startup.sh`|Alternatively, starts the previous cache|
+|`fhir/build/migration/bin/4_current-migrate.sh`|Migrates the database|
+|`fhir/build/migration/bin/5_current-pre-integration-test.sh`|Starts the server with the current settings|
+|`fhir/build/migration/bin/6_current-reindex.sh`|Runs a reindex|
+|`fhir/build/migration/bin/7_current-integration-test.sh`|Starts the current integration tests|
+|`fhir/build/migration/bin/8_teardown.sh`|Shuts down the docker images and prunes the db|
+|`fhir/build/common/gather-logs.sh migration`|gathers the log files|
+
+`fhir/` is the directory the code is the destination where `main` is checked out.
+`prev/` is the directory the previous release is checked out to, which is generally a tag.
+
+The logs are in the workarea under each datastore specific set of scripts.
+
+Transaction Timeout is 300 seconds.
```

Comment:
is it 300 or 240?

Reply:
It actually doesn't matter on this line, so I removed it.  It was a highlevel design at that time, and it codified on - whatever is preset ...